### DISCUSSION TITLE
./build.sh results in "zsh: no such file or directory: ./build.sh"

### DISCUSSION THREAD

**Discussion by aerlenbach** [4v3ngR/immich-native-macos#25] [Q&A] [answered]
Attempting to adhere to the 3-step process. "immich-install.log" fails saying `/tmp/build.sh: line 46: npm: command not found`

Step 2 "run the build.sh script to create a pkg file. `./build.sh` " doesn't specify where this code should be run. when run in the Terminal, result = `zsh: no such file or directory: ./build.sh`

**Comment by phartshorn** [comment]
npm is installed with the brew command when installing dependencies. A symlink is installed in `/usr/local/bin` when brew is working as expected. On my monterey system, brew fails to create the symlink, and I must do it manually.

Are you able to provide the full immich-install.log?

**Reply by aerlenbach** [reply]
`
INFO:  install dependencies
Warning: Formula postgresql was renamed to postgresql@14.
Warning: postgresql@14 14.15 is already installed and up-to-date.
To reinstall 14.15, run:
  brew reinstall postgresql@14
Warning: pgvector 0.8.0 is already installed and up-to-date.
To reinstall 0.8.0, run:
  brew reinstall pgvector
Warning: node 23.3.0 is already installed and up-to-date.
To reinstall 23.3.0, run:
  brew reinstall node
Warning: redis 7.2.6 is already installed and up-to-date.
To reinstall 7.2.6, run:
  brew reinstall redis
Warning: ffmpeg 7.1_3 is already installed and up-to-date.
To reinstall 7.1_3, run:
  brew reinstall ffmpeg
Warning: vips 8.16.0 is already installed and up-to-date.
To reinstall 8.16.0, run:
  brew reinstall vips
Warning: wget 1.25.0 is already installed and up-to-date.
To reinstall 1.25.0, run:
  brew reinstall wget
Warning: python@3.11 3.11.11 is already installed and up-to-date.
To reinstall 3.11.11, run:
  brew reinstall python@3.11
Warning: Formula postgresql was renamed to postgresql@14.
Stopping `postgresql@14`... (might take a while)
==> Successfully stopped `postgresql@14` (label: homebrew.mxcl.postgresql@14)
==> Successfully started `postgresql@14` (label: homebrew.mxcl.postgresql@14)
Stopping `redis`... (might take a while)
==> Successfully stopped `redis` (label: homebrew.mxcl.redis)
==> Successfully started `redis` (label: homebrew.mxcl.redis)
/Users/***
INFO: configure postgresql
INFO: configure postgresql
/private/tmp/PKInstallSandbox.pjoPQ1/Scripts/com.unofficial.immich.installer.51vMI5/configurepostgres.sh: line 20: psql: command not found
INFO: create user

INFO: create paths
DEBUG: going to switch to immich user
DEBUG: copying scripts to accessible location
INFO: building immich
INFO: cloning immich repo
Cloning into '/tmp/immich-08C01CC8-62B4-4E3B-B2B3-3FDBA2A6E985'...
HEAD is now at 97c1eb728 chore: version v1.122.1
INFO: building the server
/tmp/build.sh: line 46: npm: command not found
`

username intentionally censored.

**Reply by 4v3ngR** [reply]
Are you able to open the terminal and type `which npm` as well as `export PATH`?

Further, can you try again with the latest release (v1.122.1)?

**Reply by aerlenbach** [reply]
`which npm` results in `/opt/homebrew/bin/npm` but then I ran it again and it said `nmp not found`.

`export PATH` doesn't result in anything happening. 

Tried the new release. same result.

**Comment by 4v3ngR** [comment]
Ah, I see what the issue is... the scripts assume that brew has been installed in `/usr/local/` (the default prefix - at least on my machines).

I'll investigate how to handle other install locations for brew. In the short term, you could do a checkout of the repo, edit Scripts/config.sh and include `/opt/homebrew/bin` to the path, then build the package with the top level build.sh script.

**Reply by aerlenbach** [reply]
> In the short term, you could do a checkout of the repo, edit Scripts/config.sh and include /opt/homebrew/bin to the path, then build the package with the top level build.sh script.

I'm too stupid to do that. I'll wait patiently for you to fix in an update. thanks!

**Reply by 4v3ngR** [reply, marked as answer]
Can you give this release a go: https://github.com/4v3ngR/immich-native-macos/releases/tag/v1.122.2-b

It adds the /opt/homebrew/bin directory to the PATH environment variable.

**Reply by aerlenbach** [reply]
Installation was successful!

Thanks for the quick fix!

edit: 

I've got a new problem that might be a stupid question. but where exactly do I run`./build.sh`? Because running it in terminal results in `zsh: no such file or directory: ./build.sh`
